File: web-vue/src/types/comment.ts
import type { UserInfo } from './user'

/**
 * コメントオブジェクト型（VO）
 */
export interface CommentVO {
  id: number
  userId: number
  userInfo: UserInfo
  itemId: number
  content: string
  parentId?: number            // 親コメントのID（スレッド用）
  replyToCommentId?: number    // 返信先のコメントID
  replyToUserId?: number       // 返信先のユーザーID
  replyToUserInfo?: UserInfo   // 返信先のユーザー情報
  replies?: CommentVO[]        // 返信リスト（ネストされたコメント）
  createTime: string
  updateTime: string
}

/**
 * コメント追加用DTO
 */
export interface CommentAddDTO {
  itemId: number
  content: string
  parentId?: number
  replyToCommentId?: number
  replyToUserId?: number
}

/**
 * コメント検索用DTO
 */
export interface CommentQueryDTO {
  itemId?: number
  userId?: number
  onlyParent?: boolean         // 親コメントのみを取得するかどうか
  pageNo?: number
  pageSize?: number
}

File: web-springboot/src/main/java/com/gjq/entity/Comment.java
package com.gjq.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * コメントエンティティクラス
 */
@Data
@TableName("comment")
public class Comment {
    /**
     * コメントID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * ユーザーID
     */
    private Long userId;

    /**
     * アイテムID
     */
    private Long itemId;

    /**
     * コメント内容
     */
    private String content;

    /**
     * 親コメントID（nullの場合はトップレベルコメント、それ以外は返信を意味します）
     */
    private Long parentId;
    
    /**
     * 返信先のコメントID
     */
    private Long replyToCommentId;
    
    /**
     * 返信先のユーザーID
     */
    private Long replyToUserId;

    /**
     * 作成日時
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新日時
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}